public/app/features/panel/metrics_panel_ctrl.ts

```diff
@@ -246,18 +247,13 @@ class MetricsPanelCtrl extends PanelCtrl {
         text: 'Explore',
         icon: 'gicon gicon-explore',
         shortcut: 'x',
-        href: await getExploreUrl(this.panel, this.panel.targets, this.datasource, this.datasourceSrv, this.timeSrv),
+        href:
+          config.appSubUrl +
```

Comment:
Is it possible to move this into getExploreUrl?

Reply:
Well, yes and no. We can move it to getExploreUrl, but then we have problem, because getExploreUrl is used also in [keyBindingSrv](https://github.com/grafana/grafana/blob/master/public/app/core/services/keybindingSrv.ts#L222), where `this.$location.url` remembers the the subpath. Therefore if your root url is `https://grafana/something` it returns `https://grafana/something/something`. We could add a check/if statement to renderUrl, but I am not sure if it is a better solution than this. 
```export function renderUrl(path: string, query: UrlQueryMap | undefined): string {
  if (query && Object.keys(query).length > 0) {
    path += '?' + toUrlParams(query);
  }
  if(path.includes(config.appSubUrl)) {
    return path
  }
  return config.appSubUrl + path;
}```
What do you think?

Reply:
you can use locationUtil.stripBaseFromUrl https://github.com/grafana/grafana/blob/master/public/app/features/dashboard/services/DashboardSrv.ts#L153

Reply:
👍 nice! Thanks Torkel. Everything is now moved into getExploreUrl and for key bindings I used stripBaseFromUrl.